integrated_channels/canvas/client.py

```diff
@@ -51,9 +51,102 @@ def __init__(self, enterprise_configuration):
         self.expires_at = None
 
     def create_course_completion(self, user_id, payload):  # pylint: disable=unused-argument
-        pass
+        learner_data = json.loads(payload)
+        self._create_session()
+
+        # Retrieve the Canvas user ID from the user's edx email (it is assumed that the learner's Edx
+        # and Canvas emails will match).
+        get_user_id_from_email_url = '{url_base}/api/v1/accounts/{account_id}/users?search_term={email_address}'.format(
+            url_base=self.enterprise_configuration.canvas_base_url,
+            account_id=self.enterprise_configuration.canvas_account_id,
+            email_address=user_id
+        )
+        get_users_by_email_response = self.session.get(get_user_id_from_email_url).json()
+        try:
+            canvas_user_id = get_users_by_email_response[0]['id']
+        except (KeyError, IndexError):
+            # Trying to figure out how we should handle errors here- should we catch
+            # them in the in the transmitter? Should we return 400 but not raise?
+            # If we have multiple course completions to post, a raised exception here without
+            # anything else will prevent other transmissions to happen.
```

Comment:
If the transmission is calling the client method, we could simply raise here and catch it there.
In that instance, in the transmission we'd probably want to pool the errors and raise/set a failure state at the end or just log them as we get them. It all depends on SLAs and exact job dynamics.

You also have a bunch of other spots in this method where you're raising errors so there is already precedent to just raise up to the transmitter. I think it's best to be consistent with that unless there is a concern that every user/course completion would be affected and thus worth just crashing the program.